skimage/segmentation/slic_superpixels.py

```diff
@@ -272,14 +272,14 @@ def slic(image, n_segments=100, compactness=10., max_iter=10, sigma=0,
     n_centroids = centroids.shape[0]
     segments = np.ascontiguousarray(np.concatenate(
         [centroids, np.zeros((n_centroids, image.shape[3]))],
-        axis=-1))
+        axis=-1), dtype=dtype)
 
     # Scaling of ratio in the same way as in the SLIC paper so the
     # values have the same meaning
     step = max(steps)
     ratio = 1.0 / compactness
 
-    image = np.ascontiguousarray(image * ratio, dtype=np.double)
```

Comment:
@rfezzani a question: shouldn't this `dtype` continue here as `dtype=dtype`? Or else, if I understood correctly, `image` will return with `dtype=None` (from [np.ascontiguousarray()](https://numpy.org/doc/stable/reference/generated/numpy.ascontiguousarray.html) default `dtype` value) . Is that right? Thanks for clarifying!

Reply:
That's absolutely right @alexdesiqueira, but I will set dtype explicit for caution ;)

Reply:
Done ;)